src/passes/OptimizeInstructions.cpp

```diff
@@ -1058,15 +1063,15 @@ struct OptimizeInstructions
       // nothing much to do, except for the trivial case of adding/subbing a
       // zero
       if (auto* c = binary->right->dynCast<Const>()) {
-        if (c->value.geti32() == 0) {
+        if (c->value.getInteger() == 0LL) {
           return binary->left;
         }
       }
       return nullptr;
     }
     // wipe out all constants, we'll replace with a single added one
     for (auto* c : constants) {
-      c->value = Literal(int32_t(0));
+      c->value = Literal::makeFromInt32(0, c->type);
```

Comment:
this may need to be an i64 now, I think?

Reply:
I tried make this agnostic to type of zero. May be this not perfect? Could you suggest better approach?

Reply:
Oh, no, I think it's ok as it is, sorry.